cognite/client/_api_client.py

```diff
@@ -136,7 +136,12 @@ def _do_request(self, method: str, url_path: str, **kwargs) -> requests.Response
         headers.update(kwargs.get("headers") or {})
 
         if json_payload:
-            data = _json.dumps(json_payload, default=utils._auxiliary.json_dump_default)
+            try:
+                data = _json.dumps(json_payload, default=utils._auxiliary.json_dump_default, allow_nan=False)
+            except ValueError as e:
+                raise ValueError(f"{e}. Make sure your data does not contain NaN(s) or +/- Inf!").with_traceback(
+                    e.__traceback__
+                ) from None
```

Comment:
Any particular reason for not just doing this?
```suggestion
                raise ValueError(f"{e}. Make sure your data does not contain NaN(s) or +/- Inf!") from e
```

Reply:
Just to avoid the huge "double traceback" when all I want to do is to make the JSON error more understandable.